crates/turbo-tasks/src/graph/get_children.rs

```diff
@@ -2,12 +2,14 @@ use std::{collections::HashSet, future::Future};
 
 use anyhow::Result;
 
+use super::GraphTraversalControlFlow;
+
 /// A trait that allows a graph traversal to get the children of a node.
-pub trait GetChildren<T, A = ()> {
+pub trait GetChildren<T, A = !, Impl = ()> {
     type Children: IntoIterator<Item = T>;
     type Future: Future<Output = Result<Self::Children>>;
 
-    fn get_children(&mut self, item: &T) -> Option<Self::Future>;
+    fn get_children(&mut self, item: &T) -> GraphTraversalControlFlow<Self::Future, A>;
```

Comment:
``` rs
fn join_children(&mut self, children: Children) -> GraphTraversalControlFlow<Children, A>
```

This could be called after the future completes and since it `&mut self` it can do some counting without a Mutex